src/main/java/iterator/RichIterator.java

```diff
@@ -409,7 +719,7 @@ default RichIterator<A> takeWhile(Predicate<? super A> predicate) {
      * RichIterator.apply(1,2,3,4,5,1,2,3,4,5).takeUntil(x -> x > 3) // 1,2,3
      */
     default RichIterator<A> takeUntil(Predicate<? super A> predicate) {
-        throw new NotImplementedException();
+        return takeWhile(element -> !predicate.test(element));
```

Comment:
`Predicate` has a builtin method for that.
```suggestion
        return takeWhile(predicate.negate());
```
But the code reuse is great.